### ISSUE TITLE
Using Vite 4 Modal Failing to Open

### ISSUE THREAD

**Issue by codylindley** [swordev/suid#157]
FYI Something is failing, when using Vite 4.

Click modal, nothing happens.

https://stackblitz.com/edit/angular-vmqvmy-lv8str?file=src%2FApp.tsx,package.json

But Vite Build, it works.

Change Vite to 3.2.5 and works again.

**Comment by NinoM4ster** [comment]
Can confirm, just tried updating all packages in a project and the `vite v4.0.1` + `@suid/vite-plugin v0.1.0` combination is giving me the error `Make sure your app is wrapped in a <Router />`, even though it's definitely wrapped.

`vite v3.2.5` + `@suid/vite-plugin v0.1.0` works as expected.

Here's some other console warnings that I haven't seen before this:
```
You appear to have multiple instances of Solid. This can lead to unexpected behavior.

cleanups created outside a `createRoot` or `render` will never be run

computations created outside a `createRoot` or `render` will never be disposed
```

**Comment by NinoM4ster** [comment]
Related/Duplicate of #156

**Comment by juanrgm** [comment]
Is it a SolidJS issue? (https://github.com/solidjs/solid/issues/1426)

**Comment by juanrgm** [comment]
This was fixed in Vite v4.0.2 (https://stackblitz.com/edit/angular-vmqvmy-uqqlfj?file=package.json).

**Comment by IronGeek** [comment]
@juanrgm if `@suid/vite-plugin` does work with `vite v4.0.2`, please bump the peer dependency version to:

```json
  "vite": "^3.0.0 || ^4.0.2"
```
right now it still consider vite 4 as unmet dependency

https://github.com/swordev/suid/blob/c07d2d0d7679c5430bebd94585a981f5a441b31f/packages/vite-plugin/package.json#L37-L39
